src/mysql/result_set.cr

```diff
@@ -88,6 +88,23 @@ class MySql::ResultSet < DB::ResultSet
     end
   end
 
+  def read(t : UUID.class)
+    row_packet = @row_packet.not_nil!
+    is_nil = @null_bitmap[@column_index + 2]
+    col = @column_index
+    @column_index += 1
+
+    if is_nil
+      nil
+    elsif @columns[col].flags.bits_set?(128)
```

Comment:
Although the overload for UUID.class is probably needed, the check for binary flag might be needed to go on the arg-less `read` so that those fields are returned as `Slice` probably.

From there the `null_bitmap` and the binary flag check might go away from this method.

And last, due to some type system quirks an overload for `def read(t : (UUID | Nil).class)` might also be needed. Calling `rs.read(UUID?)` es expected to work. That is something checked by sample_value.

Reply:
I agree with:

> the check for binary flag might be needed to go on the arg-less read so that those fields are returned as Slice probably.

That was the original plan, as if it comes back as a binary flag then i think we should pass back a `Slice` however, the lines:
```
      if val.is_a?(Slice(UInt8)) && @columns[col].character_set != 63
        ::String.new(val)
```

contradicts this directly. Binary columns return as the string type: `0xfeu8` but the difference is that we always convert it to string. I would think that if it is a binary `character_set` (or more importantly if it is flagged as binary) it should return a slice. However this is a breaking change in the lib. Hence the need for me to make the overload method.